tests/test_locality_NeighborQuery.py

```diff
@@ -375,87 +375,83 @@ def test_exclude_ii(self):
 
         assert ij1 == ij2
 
-    def test_exhaustive_search(self):
+    i = ("i", list(range(10)))
+
+    @pytest.mark.parametrize(*i)
+    def test_exhaustive_search(self, i):
         L, r_max, N = (10, 1.999, 32)
 
         box = freud.box.Box.cube(L)
         seed = 0
 
-        for i in range(10):
-            _, points = freud.data.make_random_system(L, N, seed=seed + i)
-            all_vectors = points[:, np.newaxis, :] - points[np.newaxis, :, :]
-            all_vectors = box.wrap(all_vectors.reshape((-1, 3))).reshape(
-                all_vectors.shape
-            )
-            all_rsqs = np.sum(all_vectors**2, axis=-1)
-            (exhaustive_i, exhaustive_j) = np.where(
-                np.logical_and(all_rsqs < r_max**2, all_rsqs > 0)
-            )
+        _, points = freud.data.make_random_system(L, N, seed=seed + i)
+        all_vectors = points[:, np.newaxis, :] - points[np.newaxis, :, :]
+        all_vectors = box.wrap(all_vectors.reshape((-1, 3))).reshape(all_vectors.shape)
+        all_rsqs = np.sum(all_vectors**2, axis=-1)
+        (exhaustive_i, exhaustive_j) = np.where(
+            np.logical_and(all_rsqs < r_max**2, all_rsqs > 0)
+        )
 
-            exhaustive_ijs = set(zip(exhaustive_i, exhaustive_j))
-            exhaustive_counts = Counter(exhaustive_i)
-            exhaustive_counts_list = [exhaustive_counts[j] for j in range(N)]
+        exhaustive_ijs = set(zip(exhaustive_i, exhaustive_j))
+        exhaustive_counts = Counter(exhaustive_i)
+        exhaustive_counts_list = [exhaustive_counts[j] for j in range(N)]
 
-            nq = self.build_query_object(box, points, r_max)
-            result = list(
-                nq.query(points, dict(mode="ball", r_max=r_max, exclude_ii=True))
-            )
-            ijs = {(x[1], x[0]) for x in result}
-            counts = Counter([x[1] for x in result])
-            counts_list = [counts[j] for j in range(N)]
-
-            try:
-                assert exhaustive_ijs == ijs
-            except AssertionError:
-                print(f"Failed neighbors, random seed: {seed} (i={i})")
-                raise
-
-            try:
-                assert exhaustive_counts_list == counts_list
-            except AssertionError:
-                print(f"Failed neighbor counts, random seed: {seed} (i={i})")
-                raise
-
-    def test_exhaustive_search_asymmetric(self):
+        nq = self.build_query_object(box, points, r_max)
+        result = list(nq.query(points, dict(mode="ball", r_max=r_max, exclude_ii=True)))
+        ijs = {(x[1], x[0]) for x in result}
+        counts = Counter([x[1] for x in result])
+        counts_list = [counts[j] for j in range(N)]
+
+        try:
+            assert exhaustive_ijs == ijs
+        except AssertionError:
+            print(f"Failed neighbors, random seed: {seed} (i={i})")
+            raise
```

Comment:
I'm pretty sure this `try/except AssertionError` is a relic of code that existed before converting to pytest. Pytest will give us all the debugging info we need (test name and parameters) so we can make this much simpler.
```suggestion
        assert exhaustive_ijs == ijs
```